syntax = "proto3";

package pdf_to_text;

// Сервис для конвертации PDF в текст и хранения в Qdrant
service PDFToTextService {
    // Конвертация PDF в текст с сохранением в Qdrant
    rpc ConvertPDF(ConvertPDFRequest) returns (ConvertPDFResponse);
    
    // Поиск по документам
    rpc SearchDocuments(SearchRequest) returns (SearchResponse);
    
    // Удаление документа
    rpc DeleteDocument(DeleteDocumentRequest) returns (DeleteDocumentResponse);
    
    // Проверка здоровья сервиса
    rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

// Запрос на конвертацию PDF
message ConvertPDFRequest {
    bytes pdf_content = 1;        // Содержимое PDF файла
    string doc_id = 2;            // ID документа (опционально, генерируется если не указан)
    string filename = 3;          // Имя файла
}

// Ответ на конвертацию PDF
message ConvertPDFResponse {
    bool success = 1;
    string doc_id = 2;
    string message = 3;
    int32 text_length = 4;
    int32 chunks_count = 5;
    int32 points_uploaded = 6;
    double processing_time = 7;
    string error = 8;
}

// Запрос на поиск
message SearchRequest {
    string query = 1;
    int32 limit = 2;
    double score_threshold = 3;
}

// Результат поиска
message SearchResult {
    string id = 1;
    double score = 2;
    string doc_id = 3;
    string text = 4;
    int32 chunk_index = 5;
    string filename = 6;
}

// Ответ на поиск
message SearchResponse {
    repeated SearchResult results = 1;
    int32 count = 2;
}

// Запрос на удаление документа
message DeleteDocumentRequest {
    string doc_id = 1;
}

// Ответ на удаление документа
message DeleteDocumentResponse {
    bool success = 1;
    string message = 2;
}

// Запрос на проверку здоровья
message HealthCheckRequest {
}

// Ответ на проверку здоровья
message HealthCheckResponse {
    string status = 1;
    string service = 2;
    string version = 3;
    bool qdrant_available = 4;
    string timestamp = 5;
}



